src/main/java/org/jenkinsci/plugins/durabletask/executors/OnceRetentionStrategy.java

```diff
@@ -49,12 +54,13 @@
 
     private transient boolean terminating;
     
-    private int idleMinutes;
+    protected int idleMinutes;
```

Comment:
I guess I am not understanding your use case. This class was not designed for subclassing.

My understanding is that you want to provide a user option in `docker`’s `Cloud` to select one of several retention strategies known to be meaningful for this cloud, right? So you could either

* Provide a fully custom configuration UI for that, in which case this PR is unnecessary because you would internally call the `OnceRetentionStrategy` constructor directly from `docker` plugin code when creating a cloud slave. `mock-slave` does this, for example.
* Or, provide a customized `List<Descriptor<RetentionStrategy<?>>>` to a `f:heteroProperty` or the like, in which case the standard `config` view of each strategy is displayed, and you would follow standard databinding idioms including a `public` getter (and `Descriptor.getDisplayName` would be used).

You seem to trying some other approach but I do not understand what that is, and I cannot really evaluate the PR without that background. (There was no description in the PR stating the use case.)